### ISSUE TITLE
Issue with different output with different torch versions.

### ISSUE THREAD

**Issue by wicaksonoleksono** [pytorch/pytorch#154411]
```
import torch
import random
import numpy as np
def set_seed(seed=42):
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    if torch.cuda.is_available():
        torch.cuda.manual_seed(seed)
        torch.cuda.manual_seed_all(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False
    print(f"seed is {seed}")
```
I already use seed control 
And in my code it has a very different output, i use a local machine to debug and unit test a pipeline 
in my local machine which is torch 2.3 with cuda 11.8 on Linux , it has a better output than in my HPC which use torch 2.7 witch 11.8 Cuda. and it has 3% less f1-score. from what i suspect this is not a data problem or data loader / seed problem because i reload my data in my local and HPC and the result is  consistent. 

I cannot use HPC anymore because it's expensive . Thanks ..

**Comment by ngimel** [comment]
We don't guarantee bitwise reproducibility across different pytorch versions https://docs.pytorch.org/docs/stable/notes/randomness.html#reproducibility.
